src/api/spec/features/beta/webui/reports_spec.rb

```diff
@@ -0,0 +1,96 @@
+require 'browser_helper'
+
+RSpec.describe 'Reports', :js, :vcr do
+  before do
+    Flipper.enable(:content_moderation)
+  end
+
+  describe 'after reporting a comment on a project' do
+    let(:user) { create(:confirmed_user, login: 'jane_doe') }
+    let(:project) { create(:project, name: 'some_random_project') }
+    let!(:comment) { create(:comment, commentable: project) }
+
+    before do
+      login user
+      visit project_show_path(project)
+    end
+
+    it 'displays the "You reported this comment." message instantly' do
+      expect(page).not_to have_text('You reported this comment.')
```

Comment:
```suggestion
```

I would not test for a page not having the `'You reported this comment'` text just after loading a project page. It looks obvious to me, that the project page won't have this text every time a project page is loaded.

The same applies in the rest of the `it` clauses below, but for packages, comments and users.

Reply:
Without checking for the absence of `'You reported this comment'` message when the page loads, how would we prove that the action of reporting a comment really triggered the message to appear?

Reply:
> Without checking for the absence of 'You reported this comment' message when the page loads, how would we prove that the action of reporting a comment really triggered the message to appear?

Showing the `'You reported this comment'` message should not happen when the page loads, ever. We shouldn't check for things that doesn't happen in a controlled environment.

> It could have been there already before if you already reported a comment on a project/package. Sure, in the specs, it doesn't because there aren't other comments.

Yes, we could write tests switching from page with notifications to other pages. But this is not the case. And I don't find it necessary in this pull request.

These tests first load a page in an environment where no other report has been submitted before. That is what I meant with a controlled environment. That should be enough for not including the `expect(page).not_to have_text('You reported this comment.')` check.

Reply:
It's true that since we fully set the page how we want it to be, it's redundant to check for the absence of the message. I'll remove those lines.